packages/cli/src/build.ts

```diff
@@ -0,0 +1,63 @@
+import * as esbuild from 'esbuild';
+import fs from 'fs';
+import path from 'path';
+
+import { resolveFile, readIntegrationManifest } from './manifest';
+
+const GITBOOK_OUTPUT_DIR = '.gitbook';
+
+interface BuildOutput {
+    /** path where the final script is outputted */
+    path: string;
+    /** the text content of built script */
+    script: string;
+    /** resolved manifest config used during the process */
+    manifest: Awaited<ReturnType<typeof readIntegrationManifest>>;
+}
+
+/**
+ * Build the script into a single worker definition.
+ */
+export async function buildScriptFromManifest(manifestSpecPath: string): Promise<BuildOutput> {
+    const manifest = await readIntegrationManifest(manifestSpecPath);
+    /**
+     * Resolve the input and output file paths relatve to the manifest file.
+     */
+    const inputFilePath = resolveFile(manifestSpecPath, manifest.script);
+    const outputFilePath = resolveFile(
+        manifestSpecPath,
+        path.join(GITBOOK_OUTPUT_DIR, 'script.js')
```

Comment:
Where is this directory stored?

It's unclear from the `path.join(GITBOOK_OUTPUT_DIR, 'script.js')` missing a root, it looks like it'll use `process.cwd()`.

Why not use a temporary directory instead of polluting the local directory?

Reply:
The directory is stored in the same parent as the `gitbook-manifest.yaml` (look  `resolveFile` method's first argument)

As for the choice of directory `.gitbook`, again I am going with other popular CLIs like NextJS that uses a `.next` folder to output files during `next dev`

Reply:
I don't think we can compare to Next, we are forcing the users to gitignore `.gitbook`.

Also `.gitbook` is a special folder when syncing with GitHub that should not be gitignored ;) so this PR will prevent the use of the GitSync.

Reply:
Oh right, good point. How about `.gitbook-dev` folder? We can generate a `.gitignore` for this folder during `gitbook create` that adds `.gitbook-dev` to gitignore